api/db/authorization.js

```diff
@@ -0,0 +1,40 @@
+const defaultBookshelf = require('bookshelf');
+
+let activitiesModel;
+let rolesModel;
+
+const activitiesModelDefinition = {
+  tableName: 'auth_activities',
+  roles() {
+    return this.belongsToMany(
+      rolesModel,
+      'auth_role_activity_mapping',
+      'activity_id',
+      'role_id'
+    );
+  }
+};
+
+const rolesModelDefinition = {
+  tableName: 'auth_roles',
+  activities() {
+    return this.belongsToMany(
+      activitiesModel,
+      'auth_role_activity_mapping',
+      'role_id',
+      'activity_id'
+    );
+  }
+};
+
+module.exports = (bookshelf = defaultBookshelf) => {
+  if (!activitiesModel) {
```

Comment:
i don't understand why these conditional checks are needed (`if (!a) a = bookshelf.extend...`)

Reply:
Because the default export a function that returns the two models, so this prevents those models from being recreated every time it's called.  Basically it's to support cacheing.  I'm not convinced this is the best way to accomplish that.  Maybe they should export a `setup` method that creates the models and the default export is just an object or something?  🤔

Reply:
Got it, thanks -- i'm not opposed to this pattern fwiw